src/lib/swapper/swappers/CowSwapper/utils/utils.ts

```diff
@@ -0,0 +1,13 @@
+import { KnownChainIds } from '@shapeshiftoss/types'
+
+import type { CowChainId } from '../types'
+
+export const isCowswapSupportedChainId = (
+  chainId: string | undefined,
+  supportedChains: KnownChainIds[],
+): chainId is CowChainId => {
+  return (
+    (chainId === KnownChainIds.EthereumMainnet || chainId === KnownChainIds.GnosisMainnet) &&
+    supportedChains.includes(chainId as CowChainId)
```

Comment:
Aren't the left and the right side doing the same check here?

Reply:
By accident, yes - however given that feature flags are likely to be removed soon, we can keep only this implementation instead of passing supported chains around everywhere. They are equal by accident if the FF is turned on, however the first condition checks for all chains supported by the swapper, whereas the second one checks against the enabled ones. Probably `enabledChainIds` would be a better name

Reply:
If the plan is to remove the notion of `supportedChainIds` once the flag is on/removed (which we probably should, that adds a lot of boilerplate and this array being passed around, we may just bring it back if CoW ever add support for another chain we support), agree with the rationale for keeping it.

Let's keep both sides and add a comment explaining why we need this

Reply:
gotcha, added a comment

Reply:
https://github.com/shapeshift/web/commit/66611cd028621ddf2b118660f19c70ca0cd29887